# Análisis Exploratorio de Datos (EDA) – 20 Preguntas y Preguntas de Negocio

En este notebook se plantean 20 preguntas que combinan operaciones de filtrado y agrupación sobre el dataset de ventas de Adidas en EE.UU. Además, se incluyen preguntas de negocio para que los estudiantes deban abstraer y razonar sobre las implicaciones comerciales de los datos.

El ejercicio abarca:
- Análisis descriptivo básico (dimensiones, unidades de observación).
- Cálculo de totales, promedios y márgenes.
- Agrupaciones por Retailer, Estado, Producto y Método de Venta.
- Filtrados para extraer subconjuntos de datos.
- Preguntas de negocio para interpretar resultados y proponer soluciones basadas en la información.

¡Comienza resolviendo cada punto y analizando los resultados obtenidos!

In [2]:
!wget https://github.com/prayagpadwal/Adidas-Sales-Analysis/raw/main/Adidas%20US%20Sales%20Datasets.xlsx

--2025-03-06 21:02:07--  https://github.com/prayagpadwal/Adidas-Sales-Analysis/raw/main/Adidas%20US%20Sales%20Datasets.xlsx
Resolving github.com (github.com)... 140.82.112.3
Connecting to github.com (github.com)|140.82.112.3|:443... connected.
HTTP request sent, awaiting response... 302 Found
Location: https://raw.githubusercontent.com/prayagpadwal/Adidas-Sales-Analysis/main/Adidas%20US%20Sales%20Datasets.xlsx [following]
--2025-03-06 21:02:07--  https://raw.githubusercontent.com/prayagpadwal/Adidas-Sales-Analysis/main/Adidas%20US%20Sales%20Datasets.xlsx
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 693038 (677K) [application/octet-stream]
Saving to: ‘Adidas US Sales Datasets.xlsx’

Adidas US Sales Dat 100%[===================>] 676.79K  --.-KB/s    in 0.0

In [3]:
# Importar librerías necesarias
import pandas as pd
import numpy as np

# Leer el archivo Excel (asegúrate de haber subido 'Adidas US Sales Datasets.xlsx' a tu entorno de Colab)
data = pd.read_excel('Adidas US Sales Datasets.xlsx')

## Pregunta 1
**¿Cuántas filas y columnas tiene el dataset?**

_Muestra las dimensiones del DataFrame._

In [4]:
pd.DataFrame(data).shape

(9648, 13)

La tabla tiene 9648 filas y 13 columnas.

## Pregunta 2
**¿Cuál es la unidad de observación del dataset?**

_Describe qué representa cada fila (por ejemplo, una transacción o venta individual)._

In [55]:
pd.DataFrame(data).head()

Retailer  Retailer ID Invoice Date     Region     State      City  \
0  Foot Locker      1185732   2020-01-01  Northeast  New York  New York   
1  Foot Locker      1185732   2020-01-02  Northeast  New York  New York   
2  Foot Locker      1185732   2020-01-03  Northeast  New York  New York   
3  Foot Locker      1185732   2020-01-04  Northeast  New York  New York   
4  Foot Locker      1185732   2020-01-05  Northeast  New York  New York   

                     Product  Price per Unit  Units Sold  Total Sales  \
0      Men's Street Footwear            50.0        1200     600000.0   
1    Men's Athletic Footwear            50.0        1000     500000.0   
2    Women's Street Footwear            40.0        1000     400000.0   
3  Women's Athletic Footwear            45.0         850     382500.0   
4              Men's Apparel            60.0         900     540000.0   

   Operating Profit  Operating Margin Sales Method  
0          300000.0              0.50     In-store  
1          150000.0              0.30     In-store  
2          140000.0              0.35     In-store  
3          133875.0              0.35     In-store  
4          162000.0              0.30     In-store

## Pregunta 3
**¿Cuáles fueron las ventas totales?**

_Suma de la columna 'Total Sales'._

In [6]:
data["Total Sales"].sum()

899902125.0

## Pregunta 4
**¿Cuál fue la ganancia operativa total?**

_Suma de la columna 'Operating Profit'._

In [7]:
data["Operating Profit"].sum()

332134761.45000005

## Pregunta 5
**¿Cuál es el margen operativo promedio?**

_Calcula la media de la columna 'Operating Margin' y explica por qué se utiliza el promedio en lugar de una suma._

In [8]:
data["Operating Margin"].mean()

0.42299129353233833

## Pregunta 6
**Agrupar por Retailer y calcular:**
- La suma de 'Total Sales'
- La suma de 'Operating Profit'
- La suma de 'Units Sold'

_Utiliza `groupby` para obtener estos totales por cada Retailer._

In [10]:
data.groupby("Retailer")[["Total Sales","Operating Profit", "Units Sold"]].sum()

Total Sales  Operating Profit  Units Sold
Retailer                                                
Amazon          77698912.0       28818503.31      197990
Foot Locker    220094720.0       80722124.81      604369
Kohl's         102114753.0       36811252.58      287375
Sports Direct  182470997.0       74332954.96      557640
Walmart         74558410.0       25782052.61      206225
West Gear      242964333.0       85667873.18      625262

## Pregunta 7
**Calcular el margen total por Retailer y el margen operativo promedio por Retailer.**

- Margen total: (Suma de 'Operating Profit' / Suma de 'Total Sales') por Retailer.
- Margen promedio: Promedio de la columna 'Operating Margin' para cada Retailer.

_Compara ambos resultados y comenta la diferencia, reflexionando sobre el significado en términos de negocio._

In [113]:
MG=data.groupby("Retailer")[["Operating Profit", "Total Sales"]].sum()

In [114]:
MG

Operating Profit  Total Sales
Retailer                                    
Amazon              28818503.31   77698912.0
Foot Locker         80722124.81  220094720.0
Kohl's              36811252.58  102114753.0
Sports Direct       74332954.96  182470997.0
Walmart             25782052.61   74558410.0
West Gear           85667873.18  242964333.0

In [118]:
df["margen"]=df["Operating Profit"]/df["Total Sales"]

In [119]:
df

Operating Profit  Total Sales    margen
Retailer                                              
Amazon              28818503.31   77698912.0  0.370900
Foot Locker         80722124.81  220094720.0  0.366761
Kohl's              36811252.58  102114753.0  0.360489
Sports Direct       74332954.96  182470997.0  0.407369
Walmart             25782052.61   74558410.0  0.345797
West Gear           85667873.18  242964333.0  0.352594

## Pregunta 8
**¿Qué Retailer vendió la mayor cantidad de unidades totales?**

_Agrupa por Retailer, suma 'Units Sold' y ordena de forma descendente._

In [25]:
cantidad=data.groupby("Retailer")[["Units Sold"]].sum()

In [26]:
cantidad

Units Sold
Retailer                 
Amazon             197990
Foot Locker        604369
Kohl's             287375
Sports Direct      557640
Walmart            206225
West Gear          625262

In [28]:
cantidad.sort_values("Units Sold", ascending=False).head()

Units Sold
Retailer                 
West Gear          625262
Foot Locker        604369
Sports Direct      557640
Kohl's             287375
Walmart            206225

## Pregunta 9
**¿Qué producto vendió la mayor cantidad de unidades?**

_Agrupa por 'Product' y suma 'Units Sold' para identificar el producto líder._

In [31]:
mayor_unidades= data.groupby("Product")[["Units Sold"]].sum()

In [32]:
mayor_unidades

Units Sold
Product                              
Men's Apparel                  306683
Men's Athletic Footwear        435526
Men's Street Footwear          593320
Women's Apparel                433827
Women's Athletic Footwear      317236
Women's Street Footwear        392269

In [128]:
producto=mayor_unidades.sort_values("Units Sold", ascending=False).head()

In [129]:
producto

Units Sold
Product                              
Men's Street Footwear          593320
Men's Athletic Footwear        435526
Women's Apparel                433827
Women's Street Footwear        392269
Women's Athletic Footwear      317236

## Pregunta 10
**Agrupar por el método de venta ('Sales Method') y calcular:**
- La suma de 'Total Sales'
- El promedio de 'Total Sales'

_Esta agrupación ayudará a identificar cuál método tiene mejores resultados en términos de volumen y eficiencia._

In [53]:
data.groupby("Sales Method")[["Total Sales"]].sum()

Total Sales
Sales Method             
In-store      356643750.0
Online        247672882.0
Outlet        295585493.0

In [54]:
data.groupby("Sales Method")[["Total Sales"]].mean()

Total Sales
Sales Method               
In-store      204967.672414
Online         50659.210882
Outlet         97908.411063

## Pregunta 11
**¿Qué método de venta tiene mayores ventas totales y mayores ventas promedio?**

_Analiza los resultados obtenidos en la Pregunta 10 y comenta posibles razones de negocio detrás de estos resultados._

In [56]:
data.groupby("Sales Method")["Total Sales"].agg(["sum", "mean"])

sum           mean
Sales Method                            
In-store      356643750.0  204967.672414
Online        247672882.0   50659.210882
Outlet        295585493.0   97908.411063

## Pregunta 12
**Agrupar por State y calcular la suma de 'Total Sales'.**

_Determina cuál es el estado con mayores ventas totales._

In [57]:
state=data.groupby("State")[["Total Sales"]].sum()

In [74]:
state

Total Sales
State                      
Alabama          17633424.0
Alaska           14753103.0
Arizona          15782221.0
Arkansas         12639347.0
California       60174133.0
Colorado         20996536.0
Connecticut      11573448.0
Delaware         12298412.0
Florida          59283714.0
Georgia          18997466.0
Hawaii           22282457.0
Idaho            19276878.0
Illinois          9797488.0
Indiana           8836198.0
Iowa              7424011.0
Kansas            9972864.0
Kentucky         10072848.0
Louisiana        23750781.0
Maine             9195978.0
Maryland          7758059.0
Massachusetts    11076810.0
Michigan         18625433.0
Minnesota         7378068.0
Mississippi      15591709.0
Missouri          9683265.0
Montana          15710886.0
Nebraska          5929038.0
Nevada           20858509.0
New Hampshire    16411667.0
New Jersey        9934246.0
New Mexico       19865016.0
New York         64229039.0
North Carolina   23956531.0
North Dakota      7735580.0
Ohio             18484583.0
Oklahoma         10620734.0
Oregon           21349674.0
Pennsylvania     10357519.0
Rhode Island      8446806.0
South Carolina   29285637.0
South Dakota      8495576.0
Tennessee        18067440.0
Texas            46359746.0
Utah              9696420.0
Vermont          14352923.0
Virginia         21575040.0
Washington       26330718.0
West Virginia    10689160.0
Wisconsin         7727469.0
Wyoming          18577517.0

In [76]:
state.sort_values("Total Sales", ascending=False).head()

Total Sales
State                      
New York         64229039.0
California       60174133.0
Florida          59283714.0
Texas            46359746.0
South Carolina   29285637.0

## Pregunta 13 (Negocio)
**¿Qué estado muestra un mejor desempeño operativo?**

_Agrupa por State y calcula el promedio de 'Operating Margin' para evaluar cuál estado opera de manera más eficiente. Comenta posibles implicaciones de negocio._

In [77]:
state1=data.groupby("State")["Operating Margin"].mean()

In [78]:
state1

State
Alabama           0.553565
Alaska            0.349861
Arizona           0.389815
Arkansas          0.440509
California        0.386852
Colorado          0.427083
Connecticut       0.401620
Delaware          0.427986
Florida           0.390222
Georgia           0.413704
Hawaii            0.319583
Idaho             0.458935
Illinois          0.449514
Indiana           0.433889
Iowa              0.420903
Kansas            0.411667
Kentucky          0.448611
Louisiana         0.446852
Maine             0.398889
Maryland          0.407431
Massachusetts     0.347130
Michigan          0.475764
Minnesota         0.408125
Mississippi       0.454306
Missouri          0.453194
Montana           0.471042
Nebraska          0.457917
Nevada            0.405926
New Hampshire     0.391019
New Jersey        0.427083
New Mexico        0.392917
New York          0.408333
North Carolina    0.463542
North Dakota      0.398611
Ohio              0.459028
Oklahoma          0.435093
Oregon            0.413611
Pennsylvania      0.425880
Rhode Island      0.412361
South Carolina    0.433056
South Dakota      0.394653
Tennessee         0.506574
Texas             0.449144
Utah              0.444676
Vermont           0.444259
Virginia          0.414398
Washington        0.325069
West Virginia     0.456806
Wisconsin         0.424236
Wyoming           0.402778
Name: Operating Margin, dtype: float64

In [126]:
estados=state1.sort_values(ascending=False).head()

In [127]:
estados

State
Alabama           0.553565
Tennessee         0.506574
Michigan          0.475764
Montana           0.471042
North Carolina    0.463542
Name: Operating Margin, dtype: float64

## Pregunta 14
**Filtro 1: Crea un DataFrame filtrado para el Retailer con mayores ventas totales.**

_Utiliza la agrupación previa para identificar dicho Retailer y filtra el dataset original._

In [64]:
Mayores_Ventas=data.groupby("Retailer")[["Total Sales"]].sum()

In [65]:
Mayores_Ventas

Total Sales
Retailer                  
Amazon          77698912.0
Foot Locker    220094720.0
Kohl's         102114753.0
Sports Direct  182470997.0
Walmart         74558410.0
West Gear      242964333.0

In [66]:
Mayores_Ventas.sort_values("Total Sales", ascending=False).head()

Total Sales
Retailer                  
West Gear      242964333.0
Foot Locker    220094720.0
Sports Direct  182470997.0
Kohl's         102114753.0
Amazon          77698912.0

In [85]:
Mayores_Ventas=data[data['Retailer']== "West Gear"]

In [86]:
Mayores_Ventas

Retailer  Retailer ID Invoice Date     Region       State  \
140   West Gear      1128299   2020-11-01      South       Texas   
141   West Gear      1128299   2020-11-02      South       Texas   
142   West Gear      1128299   2020-11-03      South       Texas   
143   West Gear      1128299   2020-11-04      South       Texas   
144   West Gear      1128299   2020-11-05       West  California   
...         ...          ...          ...        ...         ...   
9265  West Gear      1197831   2021-10-21      South    Oklahoma   
9554  West Gear      1185732   2021-10-12  Northeast     Vermont   
9555  West Gear      1185732   2021-11-11  Northeast     Vermont   
9556  West Gear      1185732   2021-11-11  Northeast     Vermont   
9557  West Gear      1185732   2021-11-11  Northeast     Vermont   

               City                    Product  Price per Unit  Units Sold  \
140         Houston      Men's Street Footwear            40.0         775   
141         Houston    Men's Athletic Footwear            50.0         775   
142         Houston    Women's Street Footwear            50.0         775   
143         Houston  Women's Athletic Footwear            50.0         625   
144   San Francisco              Men's Apparel            55.0         575   
...             ...                        ...             ...         ...   
9265  Oklahoma City              Men's Apparel            22.0          68   
9554     Burlington            Women's Apparel            54.0          90   
9555     Burlington      Men's Street Footwear            54.0         140   
9556     Burlington    Men's Athletic Footwear            47.0          98   
9557     Burlington    Women's Street Footwear            46.0         102   

      Total Sales  Operating Profit  Operating Margin Sales Method  
140      310000.0         108500.00              0.35       Outlet  
141      387500.0          77500.00              0.20       Outlet  
142      387500.0         135625.00              0.35       Outlet  
143      312500.0          93750.00              0.30       Outlet  
144      316250.0         158125.00              0.50       Outlet  
...           ...               ...               ...          ...  
9265       1496.0            568.48              0.38       Outlet  
9554       4860.0           2235.60              0.46       Outlet  
9555       7560.0           3780.00              0.50       Outlet  
9556       4606.0           2256.94              0.49       Outlet  
9557       4692.0           1736.04              0.37       Outlet  

[2374 rows x 13 columns]

## Pregunta 15
**Filtro 2: Crea un DataFrame filtrado para el estado con mayores ventas totales.**

_Utiliza la agrupación por State para identificar dicho estado y filtra el dataset original._

In [123]:
ventas_totales=data.groupby("State")[["Total Sales"]].sum().sort_values("Total Sales", ascending=False).head()

In [124]:
ventas_totales

Total Sales
State                      
New York         64229039.0
California       60174133.0
Florida          59283714.0
Texas            46359746.0
South Carolina   29285637.0

In [89]:
data[data['State']== "New York"]

Retailer  Retailer ID Invoice Date     Region     State      City  \
0       Foot Locker      1185732   2020-01-01  Northeast  New York  New York   
1       Foot Locker      1185732   2020-01-02  Northeast  New York  New York   
2       Foot Locker      1185732   2020-01-03  Northeast  New York  New York   
3       Foot Locker      1185732   2020-01-04  Northeast  New York  New York   
4       Foot Locker      1185732   2020-01-05  Northeast  New York  New York   
...             ...          ...          ...        ...       ...       ...   
7843        Walmart      1185732   2020-07-18  Northeast  New York  New York   
7844  Sports Direct      1197831   2020-07-19  Northeast  New York  New York   
7845  Sports Direct      1197831   2020-07-20  Northeast  New York  New York   
7846  Sports Direct      1197831   2020-07-21  Northeast  New York  New York   
7847  Sports Direct      1197831   2020-07-22  Northeast  New York  New York   

                        Product  Price per Unit  Units Sold  Total Sales  \
0         Men's Street Footwear            50.0        1200     600000.0   
1       Men's Athletic Footwear            50.0        1000     500000.0   
2       Women's Street Footwear            40.0        1000     400000.0   
3     Women's Athletic Footwear            45.0         850     382500.0   
4                 Men's Apparel            60.0         900     540000.0   
...                         ...             ...         ...          ...   
7843            Women's Apparel            48.0         340      16320.0   
7844      Men's Street Footwear            20.0         270       5400.0   
7845    Men's Athletic Footwear            26.0         270       7020.0   
7846    Women's Street Footwear            27.0         238       6426.0   
7847  Women's Athletic Footwear            27.0         210       5670.0   

      Operating Profit  Operating Margin Sales Method  
0            300000.00              0.50     In-store  
1            150000.00              0.30     In-store  
2            140000.00              0.35     In-store  
3            133875.00              0.35     In-store  
4            162000.00              0.30     In-store  
...                ...               ...          ...  
7843           4896.00              0.30       Outlet  
7844           2106.00              0.39       Outlet  
7845           2597.40              0.37       Outlet  
7846           2506.14              0.39       Outlet  
7847           2948.40              0.52       Outlet  

[360 rows x 13 columns]

## Pregunta 16
**Filtro 3: Crea un DataFrame filtrado para el producto con mayor cantidad de unidades vendidas.**

_Identifica el producto líder y extrae las transacciones relacionadas._

In [91]:
data.groupby("Product")[["Units Sold"]].sum().sort_values("Units Sold", ascending=False).head()

Units Sold
Product                              
Men's Street Footwear          593320
Men's Athletic Footwear        435526
Women's Apparel                433827
Women's Street Footwear        392269
Women's Athletic Footwear      317236

In [92]:
data[data['Product']== "Men's Street Footwear"]

Retailer  Retailer ID Invoice Date     Region          State  \
0     Foot Locker      1185732   2020-01-01  Northeast       New York   
6     Foot Locker      1185732   2020-01-07  Northeast       New York   
12    Foot Locker      1185732   2020-01-25  Northeast       New York   
18    Foot Locker      1185732   2020-01-31  Northeast       New York   
24    Foot Locker      1185732   2020-02-06  Northeast       New York   
...           ...          ...          ...        ...            ...   
9621  Foot Locker      1185732   2021-10-19  Northeast  New Hampshire   
9627  Foot Locker      1185732   2021-11-18  Northeast  New Hampshire   
9633  Foot Locker      1185732   2021-12-17  Northeast  New Hampshire   
9639  Foot Locker      1185732   2021-01-24  Northeast  New Hampshire   
9645  Foot Locker      1185732   2021-02-22  Northeast  New Hampshire   

            City                Product  Price per Unit  Units Sold  \
0       New York  Men's Street Footwear            50.0        1200   
6       New York  Men's Street Footwear            50.0        1250   
12      New York  Men's Street Footwear            50.0        1220   
18      New York  Men's Street Footwear            50.0        1200   
24      New York  Men's Street Footwear            60.0        1220   
...          ...                    ...             ...         ...   
9621  Manchester  Men's Street Footwear            52.0         162   
9627  Manchester  Men's Street Footwear            59.0         149   
9633  Manchester  Men's Street Footwear            49.0         200   
9639  Manchester  Men's Street Footwear            36.0         165   
9645  Manchester  Men's Street Footwear            41.0         184   

      Total Sales  Operating Profit  Operating Margin Sales Method  
0        600000.0         300000.00              0.50     In-store  
6        625000.0         312500.00              0.50     In-store  
12       610000.0         305000.00              0.50       Outlet  
18       600000.0         300000.00              0.50       Outlet  
24       732000.0         366000.00              0.50       Outlet  
...           ...               ...               ...          ...  
9621       8424.0           3538.08              0.42       Outlet  
9627       8791.0           3780.13              0.43       Outlet  
9633       9800.0           4606.00              0.47       Outlet  
9639       5940.0           2197.80              0.37       Outlet  
9645       7544.0           2791.28              0.37       Outlet  

[1610 rows x 13 columns]

## Pregunta 17 (Negocio)
**¿Qué Retailer presenta el mayor margen operativo y por qué podría ser significativo desde el punto de vista del negocio?**

_Analiza el resumen por Retailer y discute las posibles implicaciones (por ejemplo, eficiencia operativa, estrategia de precios, etc.)._

In [120]:
df

Operating Profit  Total Sales    margen
Retailer                                              
Amazon              28818503.31   77698912.0  0.370900
Foot Locker         80722124.81  220094720.0  0.366761
Kohl's              36811252.58  102114753.0  0.360489
Sports Direct       74332954.96  182470997.0  0.407369
Walmart             25782052.61   74558410.0  0.345797
West Gear           85667873.18  242964333.0  0.352594

Sports Direct se destaca como líder en la rentabilidad, ya que tiene un márgen alto a diferencia de los demás retailers, esto se debe a que tienen una eficiencia de costos buena pero West Gear a pesar de no tener un margen alto, presenta un buen margen de ganancias y buenas ventas totales, esta empresa vende por volúmen.

## Pregunta 18 (Negocio)
**Analiza el desempeño entre métodos de venta (por ejemplo, In-store vs. Online si aplica).**

_Si el dataset incluye ambos métodos, comenta sobre las posibles diferencias en ventas totales y promedios y qué implicaciones podrían tener en la estrategia comercial._

In [121]:
datase=data.groupby("Sales Method")[["Total Sales", "Units Sold"]].sum()

In [122]:
datase

Total Sales  Units Sold
Sales Method                         
In-store      356643750.0      689990
Online        247672882.0      939093
Outlet        295585493.0      849778

## Pregunta 19
**Filtro Combinado: Crea un DataFrame filtrado que cumpla con los siguientes criterios:**
- Retailer con mayores ventas totales
- Estado con mayores ventas totales
- Producto con mayor cantidad de unidades vendidas
- Método de venta con mayores ventas totales

_Muestra las primeras filas del DataFrame resultante._

In [138]:
data[(data["Retailer"]== "West Gear")&(data["State"]== "New York")&(data["Product"]=="Men's Street Footwear")&(data["Sales Method"]== "In-Store")]


Empty DataFrame
Columns: [Retailer, Retailer ID, Invoice Date, Region, State, City, Product, Price per Unit, Units Sold, Total Sales, Operating Profit, Operating Margin, Sales Method]
Index: []

## Pregunta 20 (Negocio)
**Basado en el análisis de las variables (ventas, margen, unidades vendidas) y la matriz de correlación (si se llegase a calcular en un análisis posterior), ¿cuáles crees que son los principales factores que contribuyen a la ganancia operativa?**

_Esta pregunta es abierta: reflexiona sobre la relación entre precio por unidad, unidades vendidas, ventas totales y margen operativo, y plantea posibles estrategias de negocio basadas en estos insights._

## Conclusión

En este notebook se han planteado 20 preguntas que combinan análisis técnico y reflexiones de negocio. Se ha trabajado con filtrados y agrupaciones para extraer insights clave del dataset de ventas, lo que permite a los estudiantes practicar y comprender cómo los datos pueden informar decisiones estratégicas en un entorno comercial.